cristin-project/src/main/java/no/unit/nva/cristin/projects/CristinProjectBuilder.java

```diff
@@ -79,10 +91,31 @@ private CristinPerson getCristinPerson(NvaContributor contributor) {
         if (getCristinRole(contributor.getType()).isPresent()) {
             cristinRole.setRoleCode(getCristinRole(contributor.getType()).get());
         }
-        cristinRole.setInstitution(toCristinInstitution(contributor.getAffiliation()));
+
+        String unitIdentifier = extractUnitIdentifierIfPresent(contributor);
+
+        if (nonNull(unitIdentifier)) {
```

Comment:
If unit identifier is present, set unit to role instead of institution

Reply:
I feel that the need to comment means that the code should be more descriptive, because the comments here are not going to be accessible when we will be refactoring the code.